.github/workflows/docker-build-push-dockerhub.yml

```diff
@@ -112,6 +114,43 @@ jobs:
       #   run: |
       #     DOCKER_BUILDKIT=0 docker compose build --build-arg DATABASE_URL=postgresql://${{ env.POSTGRES_USER }}:${{ env.POSTGRES_PASSWORD }}@${{ env.DATABASE_HOST }}/${{ env.POSTGRES_DB }} calcom
 
+      - name: Test runtime
+        run: |
+            docker run --rm --network stack \
+              -p 3000:3000 \
+              -e DATABASE_URL=postgresql://${{ env.POSTGRES_USER }}:${{ env.POSTGRES_PASSWORD }}@database/${{ env.POSTGRES_DB }} \
+              -e NEXTAUTH_SECRET=${{ env.NEXTAUTH_SECRET }} \
+              -e CALENDSO_ENCRYPTION_KEY=${{ env.CALENDSO_ENCRYPTION_KEY }} \
+              docker.io/calcom/cal.com:add-runtime-check &
+              
+            server_pid=$!
+
+
+            echo "Waiting for the server to start..."
+            sleep 120
+
+            echo ${{ env.NEXT_PUBLIC_WEBAPP_URL }}/auth/login
+        
+            for i in {1..60}; do
+              echo "Checking server health ($i/60)..."
+              response=$(curl -o /dev/null -s -w "%{http_code}" ${{ env.NEXT_PUBLIC_WEBAPP_URL }}/auth/login)
+              echo "HTTP Status Code: $response"
+              if [[ "$response" == "200" ]] || [[ "$response" == "307" ]]; then
+                echo "Server is healthy"
+                # Now, shutdown the server
+                kill $server_pid
+                exit 0
+              fi
+              sleep 1
+            done
+        
+            echo "Server health check failed"
+            kill $server_pid
+            exit 1
+        env:
+          NEXTAUTH_SECRET: 'EI4qqDpcfdvf4A+0aQEEx8JjHxHSy4uWiZw/F32K+pA='
+          CALENDSO_ENCRYPTION_KEY: '0zfLtY99wjeLnsM7qsa8xsT+Q0oSgnOL'
```

Comment:
I'm not entirely sure if it's better to pass these from `env.<...>` but this works well.

Reply:
I think it's probably save since these are just arbitrary test values in this case